packages/kbn-esql-utils/src/utils/index.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+
+// Some applications need to have a dataview to work properly with ES|QL queries
+// This is a helper to create one. The id is constructed from the indexpattern.
+// As there are no runtime fields or field formatters or default time fields
+// the same adhoc dataview can be constructed/used. This comes with great advantages such
+// as solving the problem descibed here https://github.com/elastic/kibana/issues/168131
+export async function getESQLAdHocDataview(
+  indexPattern: string,
+  dataViewsService: DataViewsPublicPluginStart
+) {
+  return await dataViewsService.create({
+    title: indexPattern,
+    id: `esql-${indexPattern}`,
```

Comment:
We need to see if there's a max size for the id field. I've seen index patterns that are about 3k characters long

Reply:
Also seems like this functionality could be part of the data views service

Reply:
This is a good idea!

Reply:
I'm currently uncertain as to whether this makes sense as part of the data views api. We need to determine whether we're solving problems for just discover / dashboards /visualizations or all of kibana.

Reply:
The id needs to use a hash and the type should be set to an 'esql' constant

Reply:
ah, that's why you were adding just the type, didn't see it, however, let's go with the string, and do the type in a follow up, with also taking care of the new type on data view level